### PR TITLE
WS2-1523, WS2-1518 [install profile] - Added Sitemap XML base URL for…

### PR DESCRIPTION
**Pull request by bjdevil21**
…m; Refactored existing installation tasks; Other UX improvements

Manually rolled in WS2-1518 changes by tbutterf

### CODE REVIEW

**Comment by chill5-0** [comment]
Add a "Rebuild queue and generate" function call on install to auto generate sitemap.xml.

**Comment by bjdevil21** [comment]
@chill5-0 - I’ve added a snippet of code to force the generation at the submitForm stage, and it does get further in the process (it shows "1 of 1" generated now). But the sitemap.xml page still isn’t there after installation completion. During step-by-step debugging, I found a hidden message in the module's obscure logging that (generally) said, “The sitemap generation failed to finish. Please run cron again to finish the generation process.”

In short, the sitemap.xml page is only generated only after simple_sitemap_cron is fired again **after** the installation is finished, so this is as close as I can get. :)

**Comment by chill5-0** [comment]
@bjdevil21 Sounds good - what do you think about adding that info as help text for the domain field? That way users are aware of the additional step to generate the sitemap.

**Comment by bjdevil21** [comment]
@chill5-0 - I added some explanatory text to the form, explaining what they needed to do after the install was complete. (I didn't want to put a link in the info, just in case someone clicked on it and screwed up the install.)

`'<p><strong>NOTE:</strong> After this installation process is complete, go to /admin/config/system/cron ' .
  'and "Run Cron" once so that this change takes effect immediately.</p>',`

**Comment by chill5-0** [comment]
@bjdevil21 On test spin up (https://dev-280-jan19.ws.asu.edu/) there was an error on the Base url install page - added you as admin on that site to see the logs for ttroubleshooting.

**Review by chill5-0 — APPROVED** [review]
Ran test install and new note is there, no errors, and once cron is run, the default xml is showing up with the correct domain name.

File: webspark.profile [hunk 1]
```diff
@@ -45,17 +73,9 @@ function webspark_form_install_configure_submit($form, FormStateInterface $form_
...
-    'run' => INSTALL_TASK_RUN_IF_NOT_COMPLETED,
-  ];
-  $tasks['webspark_install_configure_ga_form'] = [
-    'display_name' => t('Google Analytics'),
-    'type' => 'form',
-    'function' => 'Drupal\webspark_installer_forms\Form\WebsparkConfigureGAForm',
-  ];
-}
+/**
+ * Implements hook_preprocess_template().
```
Comment (mlsamuelson) [inline webspark.profile:77]: Don't let this hold up the PR. It can merge. But perhaps this mis-matched hook implements note can be corrected in a future update.

**Review by mlsamuelson — APPROVED** [review]
